### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by krgarima**
## Change Logs:
# Description: Added Authentication and added +/- functionality in Cart

# More details:

1. Login Page -
- User has to login in order to access private routes (Wishlist, Cart). User can login using dummy username and password. When a user logins, login button changes to `log out`. User can log out using the `Log out` button.

2. 404 Page - 
- Added 404 Page not found. User will be shown this message if she/he types the wrong URL in the browser address bar.

3. Cart -
- Earlier the quantity of the items in the Cart reduced by 1 when user clicked on 'Remove from Cart' button. Now added +/- button to increase/ decrease quantity and delete the item when user clicks on 'Remove from Cart' button. Added a delete icon too when quantity is 1.

4. Errors - 
- Made some old fixes and changed Sign up pages' indentation .

5. Responsiveness - 
- Site has been made responsive.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -21,8 +24,11 @@ function App() {
           <Route path="/ProductListing" element={<ProductListing />} />
           <Route path="/Login" element={<Login />} />
           <Route path="/SignUp" element={<SignUp />} />
-          <Route path="/myCart" element={<Cart />} />
-          <Route path="/Wishlist" element={<Wishlist />} />
+          <Route path="/myCart" element={privateRoute ? <Cart /> : <Login />} />
```
Comment (haarsh24) [inline src/App.js]: Suggestion: If you are using PascalCase use it for every route path to maintain consistency.

File: src/Components/CartCard/CartCard.js [hunk 2]
```diff
@@ -25,24 +29,44 @@ export default function CartCard() {
...
-                      max="5"
-                      value={product.quantity}
-                    />
+                    <label htmlFor="quantity">
+                      Quantity:{" "}
+                      <button
+                        className="btn-changeQty"
+                        onClick={() =>
+                          dispatch({
+                            type: "REDUCE",
```
Comment (haarsh24) [inline src/Components/CartCard/CartCard.js]: You can use a better type name like: reduce cart quantity.

File: src/Components/CartCard/CartCard.js [hunk 3]
```diff
@@ -25,24 +29,44 @@ export default function CartCard() {
                 <div className="description1">
                   <p className="item-price">₹ {product.price}</p>
                   <p>
```
Comment (haarsh24) [inline src/Components/CartCard/CartCard.js:31]: Suggestion: you can destructure the product at the top so that  you don't have to write it every time

**Review by haarsh24 — APPROVED** [review]
1. Added some suggestion 
2. UI looks good
3. code is clean.

File: src/Components/CartCard/CartCard.js (hunk 2 above)
Comment (krgarima) [reply src/Components/CartCard/CartCard.js]: noted

File: src/Components/CartCard/CartCard.js (hunk 3 above)
Comment (krgarima) [reply src/Components/CartCard/CartCard.js:31]: If I remove `product` while mapping above and add destructured properties there then how will I access each `product` item which I want to pass to dispatch?

File: src/App.js (hunk 1 above)
Comment (krgarima) [reply src/App.js]: noted

File: src/Components/CartCard/CartCard.js (hunk 3 above)
Comment (haarsh24) [reply src/Components/CartCard/CartCard.js:31]: Ohh I didn't see that , you can leave it as it is then

File: src/Components/CartCard/CartCard.js [hunk 4]
```diff
@@ -25,24 +29,44 @@ export default function CartCard() {
...
+                        ) : (
+                          "-"
+                        )}
+                      </button>
+                      {product.quantity}
+                      <button
+                        className="btn-changeQty"
+                        onClick={() =>
+                          dispatch({
+                            type: "ADDED",
```
Comment (arulk8) [inline src/Components/CartCard/CartCard.js:54]: this ADDED can be move to file called actionTypes and used at any place we want

File: src/Pages/Login/Login.js [hunk 5]
```diff
@@ -1,31 +1,66 @@
...
+  const navigate = useNavigate();
+  const { userName, setUserName, password, setPassword, setLogged } =
+    useContext(AuthContext);
+  const { privateRoute, setPrivateRoute } = useContext(PrivateRouteContext);
+
+  const handleLogin = (e) => {
+    e.preventDefault();
+
+    if (userName === "adarshbalika@neog.camp" && password === "password") {
+      setPrivateRoute(true);
```
Comment (arulk8) [inline src/Pages/Login/Login.js:19]: This logic will not work for other users. you can check the local storage for auth token. If it is there set the route to private.  or set this private route during login/signup itself.

**Review by arulk8 — APPROVED** [review]

File: src/Pages/Login/Login.js (hunk 5 above)
Comment (krgarima) [reply src/Pages/Login/Login.js:19]: I have implemented only basic authentication for now using dummy username and password. I will implement authentication with tokens later after the Authentication class.

File: src/Components/CartCard/CartCard.js (hunk 4 above)
Comment (krgarima) [reply src/Components/CartCard/CartCard.js:54]: That seems like a good idea.

File: src/App.js (hunk 1 above)
Comment (krgarima) [reply src/App.js]: done

File: src/Components/CartCard/CartCard.js (hunk 2 above)
Comment (krgarima) [reply src/Components/CartCard/CartCard.js]: done

**Pull request merged**